packages/neo-one-client/src/__tests__/user/LocalUserAccountProvider.test.js

```diff
@@ -0,0 +1,1118 @@
+      address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
+      txid:
+        '0x7f48028c38117ac9e42c8e1f6f06ae027cdbb904eaf1a0bdc30c9d81694e045c',
+      vout: 1,
+    },
+    sevenNEO: {
+      asset: common.NEO_ASSET_HASH,
+      value: new BigNumber('7'),
+      address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
+      txid:
+        '0x7f48028c38117ac9e42c8e1f6f06ae027cdbb904eaf1a0bdc30c9d81694e045d',
+      vout: 2,
+    },
+    elevenGAS: {
+      asset: common.GAS_ASSET_HASH,
+      value: new BigNumber('11'),
+      address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
+      txid:
+        '0x7f48028c38117ac9e42c8e1f6f06ae027cdbb904eaf1a0bdc30c9d81694e045e',
+      vout: 3,
+    },
+  };
+  const results = {
+    fault: {
+      state: 'FAULT',
+      gasConsumed: new BigNumber('3'),
+      stack: [{ type: 'String', value: 'val' }],
+      message: 'testMessage',
+    },
+    halt: {
+      state: 'HALT',
+      gasConsumed: utils.ZERO_BIG_NUMBER,
+      stack: [{ type: 'String', value: 'val' }],
+    },
+  };
+  const witness = {
+    invocation: '02028a99826ed',
+    verification: '02028a99826ee',
+  };
+
+  let wallets = [];
+  const store = {
+    type: 'test',
+    getWallets: () => Promise.resolve(wallets),
+    saveWallet: wallet => {
+      wallets.push(wallet);
+      return Promise.resolve();
+    },
+    deleteWallet: wallet => {
+      wallets.splice(wallets.indexOf(wallet), 1);
+      return Promise.resolve();
+    },
+  };
+
+  let keystore = new LocalKeyStore({ store });
+  let provider = new NEOONEProvider({
+    mainRPCURL: 'mainURL',
+    testRPCURL: 'testURL',
+  });
+  let localUserAccountProvider = new LocalUserAccountProvider({
+    keystore,
+    provider,
+  });
+  beforeEach(() => {
+    wallets = [];
+    keystore = new LocalKeyStore({ store });
+    provider = new NEOONEProvider({
+      mainRPCURL: 'mainURL',
+      testRPCURL: 'testURL',
+    });
+    localUserAccountProvider = new LocalUserAccountProvider({
+      keystore,
+      provider,
+    });
+  });
+
+  test('getCurrentAccount', () => {
+    // $FlowFixMe
+    keystore.getCurrentAccount = jest.fn(() => account1);
+
+    const result = localUserAccountProvider.getCurrentAccount();
+    expect(result).toEqual(account1);
+  });
+
+  test('getAccounts', () => {
+    const expected = [account1];
+    // $FlowFixMe
+    keystore.getAccounts = jest.fn(() => expected);
+
+    const result = localUserAccountProvider.getAccounts();
+    expect(result).toEqual(expected);
+  });
+
+  test('getNetworks', () => {
+    const expected = '10';
+    // $FlowFixMe
+    provider.getNetworks = jest.fn(() => expected);
+
+    const result = localUserAccountProvider.getNetworks();
+    expect(result).toEqual(expected);
+  });
+
+  test('selectAccount', async () => {
+    // $FlowFixMe
+    keystore.selectAccount = jest.fn(() => Promise.resolve());
+
+    await localUserAccountProvider.selectAccount(id1);
+    expect(keystore.selectAccount).toHaveBeenCalledTimes(1);
```

Comment:
I like to use snapshot tests for these, something like `expect(keystore.selectAccount.mock.call).toMatchSnapshot()`